test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTxnWithFailoverTest.java

```diff
@@ -0,0 +1,444 @@
+        controllerInstance.scaleService(2, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(60000);
+        log.info("Scaling down controller instances from 3 to 2");
+
+        currentWriteCount2 = eventWriteCount.get();
+        currentReadCount2 = eventReadCount.get();
+
+        log.info("Read count after controller failover after sleep  {}", currentReadCount2);
+        log.info("Write count after controller failover after sleep  {}", currentWriteCount2);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount2 > currentWriteCount1);
+        //ensure reads are happening
+        assertTrue(currentReadCount2 > currentReadCount1);
+
+        //Scale down SSS, controller to 1 instance each.
+        segmentStoreInstance.scaleService(1, true);
+        controllerInstance.scaleService(1, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(new Random().nextInt(50000) + 60000);
+        log.info("Scaling down  to 1 controller, 1 SSS instance");
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Stop write flag status {}", stopWriteFlag);
+        log.info("Stop read flag status {}", stopReadFlag);
+        log.info("Event data {}", eventData.get());
+        log.info("Read count {}", eventReadCount.get());
+        log.info("Read count with SSS and controller failover after sleep {}", currentReadCount1);
+        log.info("Write count with SSS and controller failover after sleep {} ", currentWriteCount1);
+    }
+
+    private CompletableFuture<Void> startWritingIntoTxn(final AtomicLong data, final EventStreamWriter<Long> writer,
+                                                 final AtomicBoolean stopWriteFlag, final AtomicLong eventWriteCount) {
+        return CompletableFuture.runAsync(() -> {
+            while (!stopWriteFlag.get()) {
+                Transaction<Long> transaction = null;
+                    try {
+                        transaction = Retry.withExpBackoff(10, 10, 20, ofSeconds(1).toMillis())
+                                .retryingOn(MultiReaderWriterTxnWithFailoverTest.TxnCreationFailedException.class)
+                                .throwingOn(RuntimeException.class)
+                                .run(() -> createTransaction(writer, stopWriteFlag));
+
+                        //each transaction has 50 events
+                        for (int j = 0; j < 50; j++) {
+                            long value = data.incrementAndGet();
+                            Thread.sleep(100);
+                            transaction.writeEvent(String.valueOf(value), value);
+                            log.debug("Writing event {} into transaction {}", value, transaction);
+                        }
+                        //commit Txn
+                        transaction.commit();
```

Comment:
so you should call txn.getStatus() and wait until the status becomes committed before adding it to the count. 
this can be scheduled in the background.